src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEParameterSymbol.cs

```diff
@@ -656,6 +676,42 @@ internal override bool IsCallerMemberName
             }
         }
 
+        internal override int CallerArgumentExpressionParameterIndex
+        {
+            get
+            {
+                const WellKnownAttributeFlags flag = WellKnownAttributeFlags.IsCallerArgumentExpression;
+
+                if (!_packedFlags.TryGetWellKnownAttribute(flag, out bool value))
+                {
+                    var discardedUseSiteInfo = CompoundUseSiteInfo<AssemblySymbol>.Discarded;
+                    bool isCallerArgumentExpression = !HasCallerLineNumberAttribute
+                        && !HasCallerFilePathAttribute
+                        && !HasCallerMemberNameAttribute
+                        && HasCallerArgumentExpressionAttribute
+                        && new TypeConversions(ContainingAssembly).HasCallerInfoStringConversion(this.Type, ref discardedUseSiteInfo);
+
+                    value = _packedFlags.SetWellKnownAttribute(flag, isCallerArgumentExpression);
+                }
+
+                if (value)
+                {
+                    var info = _moduleSymbol.Module.FindTargetAttribute(_handle, AttributeDescription.CallerArgumentExpressionAttribute);
+                    _moduleSymbol.Module.TryExtractStringValueFromAttribute(info.Handle, out var parameterName);
```

Comment:
It might make sense to move this logic into a helper method on PEModule so that it could be used by VB as well.